### ISSUE TITLE
Installation error with conda

### ISSUE THREAD

**Issue by skanwal** [umccr/gpgr#16]
Hello,

We are looking into analysing some data for mutational patterns. Currently, exploring if we can leverage gpgr for this.

Is the package platform independent? On my local system (MAC), I am getting the following issue:
```
$ conda install -c pdiakumis r-gpgr
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - r-gpgr

Current channels:

  - https://conda.anaconda.org/pdiakumis/osx-64
  - https://conda.anaconda.org/pdiakumis/noarch
  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```
 Looking through this [vignette](https://umccr.github.io/gpgr/articles/mutationalpatterns.html), seems it was tested on `Ubuntu 18.04.5 LTS`. So wanted to confirm if this is a platform or set-up issue?
 
 Thanks.

**Comment by pdiakumis** [comment]
Hey there @skanwal!

You're spot on, the conda package is only built for Linux (you can confirm that yourself in general in the built file list at https://anaconda.org/pdiakumis/r-gpgr/files).
I think the reason I didn't also support Mac conda builds was due to the difficulty in getting `signature.tools.lib` and `NNLM` (which are both required for the HRD functionality of this package) working in a conda environment.

If you are interested in MutationalPatterns functionality, you can see the wrappers used in that vignette in <https://github.com/umccr/gpgr/blob/master/R/mutationalpatterns.R>.

Another thing you could try is to directly install the `gpgr` package on your Mac with `devtools::install_github("umccr/gpgr")`. **Prior** to doing that, you should try installing NNLM (https://github.com/linxihui/NNLM). That is a bit of a pain to install but I remember that installing gfortran worked for me (using R version 4 and above). Try the top `dmg` file from <https://github.com/fxcoudert/gfortran-for-macOS/releases> (your Mac might warn you about something like not trusting that dmg file or similar - just try opening it by clicking option). 

So to summarise:

- download and install https://github.com/fxcoudert/gfortran-for-macOS/releases
- `devtools::install_github("linxihui/NNLM")`
- `devtools::install_github("umccr/gpgr")` (this might work directly without requiring the above step)

Let me know how it goes :-)

**Comment by pdiakumis** [comment]
I can confirm (with great surprise) that those instructions worked on my Mac just now (using R 4.1). Even though half the internet was downloaded, you might also want to do a `BiocManager::install("BSgenome.Hsapiens.UCSC.hg38")` to get the vignette examples working (or install one of the other `BSgenome` packages which are soft dependencies as listed in https://github.com/umccr/gpgr/blob/master/DESCRIPTION#L47).

**Comment by skanwal** [comment]
Thanks @pdiakumis for looking into this.
Can also confirm I was able to reproduce and install gpgr using the three steps on intel Mac (yay!).

I am now trying to also set it up to my new machine (M1). Facing few issues with gfortran VS R 4.1 arm64 version.
I am looking into it and will update here if I can resolve the issue (or give up on it)..

**Comment by pdiakumis** [comment]
Ah, I guess you could give the second gfortran `pkg` a go and see.
Another option would be to dockerise the conda package and use that on your Mac instead, but yeah, a bit overkill.
If you just want `MutationalPatterns`, you don't need `gpgr` at all - just play with those functions in that file I linked to above.

**Comment by skanwal** [comment]
@pdiakumis Thanks for your help. - closing this issue as the package installs without any issues on Intel Mac.
